src/Features/CSharp/Portable/OrderModifiers/OrderModifiersHelper.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.CodeAnalysis.CSharp.OrderModifiers
+{
+    internal static class OrderModifiersHelper
+    {
+        private static readonly char[] s_comma = { ',' };
+        private static readonly object s_gate = new object();
+
+        private static string s_lastValue;
+        private static Dictionary<int, int> s_lastPreferredOrder;
```

Comment:
is ReadOnlyDictionary as efficient as Dictoinary?  I want this as low overhead as possible.  I'm even considering just an int[].  However, i'd have to make it about 10k large due to where our SyntaxKinds start, and that seems somewhat unfortunate.